token/js/test/e2e-2022/tokenmeta.test.ts

```diff
@@ -0,0 +1,86 @@
+import chai, { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+
+import type { Connection, PublicKey, Signer } from '@solana/web3.js';
+import { sendAndConfirmTransaction, Keypair, SystemProgram, Transaction } from '@solana/web3.js';
+import {
+    createMint,
+    createInitializeAccountInstruction,
+    getAccountLen,
+    ExtensionType,
+    initializeMetadata,
+} from '../../src';
+import { TEST_PROGRAM_ID, newAccountWithLamports, getConnection } from '../common';
+import { MINT_SIZE } from '../../src/state/mint';
+
+const TEST_TOKEN_DECIMALS = 2;
+const EXTENSIONS = [ExtensionType.CpiGuard];
```

Comment:
How do we set the account length for token metadata? From what i understand its not fixed length
I tried reading in the rust to see how it worked, but couldn't figure it out

Reply:
Yes, the metadata is comprised of multiple strings, which are variable-length. So you have to calculate the size of the metadata object using `@solana/spl-type-length-value`